core/src/main/scala/kafka/server/DynamicBrokerConfig.scala

```diff
@@ -913,6 +915,13 @@ class DynamicMetricReporterState(brokerId: Int, config: KafkaConfig, metrics: Me
     reporters.forEach { reporter =>
       metrics.addReporter(reporter)
       currentReporters += reporter.getClass.getName -> reporter
+      val clientTelemetryReceiver = reporter match {
```

Comment:
Changes not required now, and the file shall be removed when merged with open PR: https://github.com/apache/kafka/pull/14699